src/gui/torrentcontentmodel.cpp

```diff
@@ -71,11 +71,9 @@ namespace
     public:
         using QFileIconProvider::icon;
 
-        QIcon icon(const QFileInfo &info) const override
+        QIcon icon(const QFileInfo &) const override
         {
-            Q_UNUSED(info);
-            static QIcon cached = UIThemeManager::instance()->getIcon(u"text-plain"_qs);
-            return cached;
+            return UIThemeManager::instance()->getIcon(u"text-plain"_qs);
```

Comment:
Returning a variable is still faster than searching in the Cache. Is there a compelling reason to remove such optimizations (even if they are micro-optimizations)?

Reply:
`UIThemeManager::instance()->getIcon()` caches it already.

Reply:
I mean, we used to often neglect performance optimization, especially if it looked too insignificant. But as it turned out, this can result in significant performance losses when scaling. In modern realities, we can no longer dismiss those who constantly have several hundred or even thousands of torrents.

Reply:
And do you really believe that
```c++
return cached;
```
has no performance advantages against the following?
```c++
const auto iter = m_iconCache.find(iconId);
if (iter != m_iconCache.end())
    return *iter;
```

Reply:
>But as it turned out, this can result in significant performance losses when scaling. In modern realities, we can no longer dismiss those who constantly have several hundred or even thousands of torrents.

Yes, I agree with that viewpoint. 
You can point out which change is truly troubled with performance issues and we can discuss/review it. (although I already thought about it before submitting this PR)
Also note that optimizing for performance should often be backed up with measurements, otherwise it is prone to be merely speculations.

>Is there a compelling reason to remove such optimizations (even if they are micro-optimizations)?

Caching it here is the wrong design IMO, it should be done at `UIThemeManager` *if really required* (such as a specific getter function).

>has no performance advantages against the following?

It has *minor* advantages in generic cases... do you think the *minor* difference matters in this specific case?

Then what about declaring all variables as `static` whenever possible? Wouldn't that bring out the most missing performance?

Reply:
> Then what about declaring all variables as `static` whenever possible? Wouldn't that bring out the most missing performance?

No, of course. I never suggested ideas like this. `static` is neither evil nor good. It's just a tool, like many others. If it is used for its intended purpose, then it does good, and vice versa.

Reply:
> Caching it here is the wrong design IMO, it should be done at `UIThemeManager`

Trying to perform specific optimizations at a generic logic level is really wrong design.


> (or `TorrentContentModel`) _if really required_ (such as a specific getter function).

`TorrentContentModel` wants (in existing code) this specific optimization (if any) so I think it can be transformed in class member variable. The same is about icons from `TorrentModel`.
>

> Also note that optimizing for performance should often be backed up with measurements & use-case, otherwise it is prone to be merely speculations.

The same I would say about cancellation of existing optimization.

Reply:
PR updated, now they are class variables.

>static is neither evil nor good.

IMO they are leaning on the bad side. The fact they cannot be controlled to release their memory is a downside to me (allocate/deallocate is the basic building blocks).
Another one is they might introduce thread safety issues at usage if not careful (even if their initialization has some strong atomic guarantee).

Reply:
> IMO they are leaning on the bad side. The fact they cannot be controlled to release their memory is a downside to me (allocate/deallocate are the basic building blocks for OOP paradigm).

I also prefer to use other alternatives.
I'm hooked on something else. Namely, for the fact that under the guise of "code cleanup" you reject existing optimizations (at least, what was intended for optimization) without enough justification.
If you want to clean the code from "bad" `static`, well, please just replace it with a "better" alternative. If you want to cancel optimization altogether (since you think it has more cons than pros), this is a separate task.

Reply:
Sorry, I expect to deal with unruly `static` "optimizations" in this PR so please bear with me. This is the last of it.